python/src/llvm.cc

```diff
@@ -243,8 +243,25 @@ void init_triton_llvm(py::module &&m) {
 
   m.def(
       "to_module",
-      [](mlir::ModuleOp &mod, llvm::LLVMContext &ctx) {
-        return mlir::translateModuleToLLVMIR(mod, ctx);
+      [](mlir::ModuleOp &mod, llvm::LLVMContext &ctx, const std::string triple,
+         std::string proc, std::string features) {
+        std::string error;
+        auto target = llvm::TargetRegistry::lookupTarget(triple, error);
+        llvm::TargetOptions opt;
+        // Target machine is only used to create the data layout.
+        std::unique_ptr<llvm::TargetMachine> machine{
+            target->createTargetMachine(triple, proc, features, opt,
+                                        llvm::Reloc::PIC_, std::nullopt,
+                                        llvm::CodeGenOptLevel::None)};
+        auto dl = machine->createDataLayout();
+        // Set the target data layout attribute here so that the target
+        // DataLayout gets set on the llvm:Module during mlir->llir translation.
+        // This is required because LLVM optimization passes(like InstCombine)
+        // rely on having accurate target DataLayout in the llvm:Module
+        mod->setAttr(mlir::LLVM::LLVMDialect::getDataLayoutAttrName(),
+                     mlir::StringAttr::get(mod.getContext(),
+                                           dl.getStringRepresentation()));
+        return mlir::translateModuleToLLVMIR(mod, ctx, triple);
```

Comment:
why passing the triple as module name?

Reply:
This sets the data layout attribute of the module to a dl string, not actually setting the name attribute of the module?

Reply:
isn't the 3rd operand of `translateModuleToLLVMIR` the module name?

Reply:
> Oh yeah, that's true. Triple shouldn't be used there, as the data layout attribute is already set which should be enough?
translateModuleToLLVMIR does not take triple as an argument. I will remove it.